format_code.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+if (perl -e '' 2>/dev/null)
+  then SCRIPT_DIR="$(dirname "$(perl -MCwd -le 'print Cwd::abs_path(shift)' "$0")")"
+  else SCRIPT_DIR="$(dirname "$(readlink -f "$0")")"
```

Comment:
Testing for perl seems like overkill. Since you're writing this script in bash, suggest using `BASH_SOURCE`, e.g., `SCRIPT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )`